### PR TITLE
Fix: fixed bugs in paging, filtering and sorting

### PR DESCRIPTION
**Pull request by faouziH21**
This request fixes the following:
-Filtering not working when searching for something in GeneralEntityTable 
-In the filter field: undefined was appearing as a value to be searched. Now the sort field does what it is intended to do 
-Sorting not working properly
-paging not working properly
-changed a minor translation in the German translation for "Filter"

### CODE REVIEW

File: src/Util/TableUtil.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export default class TableUtil {
           <Input
             ref={node => { searchInput = node; }}
             placeholder={`${i18n.t('GeneralEntityTable.popupSearch')} ${dataIndex}`}
-            value={`${selectedKeys[0]}`}
+            value={selectedKeys[0]}
```
Comment (marcjansen) [inline src/Util/TableUtil.tsx:36]: Hm, this says:

Type 'Key' is not assignable to type 'string | number | readonly string[] | undefined'.

I like the previous thing more.

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx [hunk 2]
```diff
@@ -657,13 +654,8 @@ export function GeneralEntityRoot<T extends BaseEntity>({
...
     }
-
-    if (!_isNil(pagination.current)) {
-      setPageCurrent(pagination.current);
-    }
-    if (!_isNil(pagination.pageSize)) {
-      setPageCurrent(pagination.pageSize);
-    }
+    setPageCurrent(pagination.current!);
+    setPageSize(pagination.pageSize!);
```
Comment (marcjansen) [inline src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx:658]: This was introduced by @ahennr in https://github.com/terrestris/shogun-admin/commit/10b2c179a0881f06548b4e088afa1fee2ee59fd4, I think. Are we sure your changes are fine? can @ahennr confirm?

**Review by marcjansen — COMMENTED** [review]
Please see the comments.

Can you please try to add unit tests?

File: src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx (hunk 2 above)
Comment (faouziH21) [reply src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx:658]: This work was in cooperation with @ahennr

File: src/Util/TableUtil.tsx (hunk 1 above)
Comment (ahennr) [reply src/Util/TableUtil.tsx:36]: is `selectedKeys[0] ?? ''` an option?

File: src/Util/TableUtil.tsx [hunk 3]
```diff
@@ -33,7 +33,7 @@ export default class TableUtil {
           <Input
             ref={node => { searchInput = node; }}
             placeholder={`${i18n.t('GeneralEntityTable.popupSearch')} ${dataIndex}`}
-            value={`${selectedKeys[0]}`}
+            value={selectedKeys[0] as string}
```
Comment (ahennr) [inline src/Util/TableUtil.tsx:36]:
```suggestion
            value={_isNil(selectedKeys[0]) ? '' : `${selectedKeys[0]}`}
```

**Review by ahennr — APPROVED** [review]

**Pull request merged**